r2i/tensorflow/frame.cc

```diff
@@ -121,12 +121,13 @@ RuntimeError Frame::Validate (int64_t dims[], int64_t num_dims) {
     return error;
   }
 
-  /* Check that channels match
-   * TODO: relate this to the input format
-   */
-  if (3 != dims[3]) {
-    error.Set (RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
-               "We only support a 3 channels per image");
+  /* Check that channels match */
+  int frame_format_channels = this->frame_format.GetNumPlanes();
```

Comment:
Move variable definitions at the top of the function. Assignment can remain here